src/common.c

```diff
@@ -462,6 +469,24 @@ static int ofi_str_to_psmx2(const char *str, void **addr, size_t *len)
 	return -FI_EINVAL;
 }
 
+static int ofi_str_to_psmx3(const char *str, void **addr, size_t *len)
+{
+	int ret;
+
+	*len = 2 * sizeof(uint64_t);
+	*addr = calloc(1, *len);
+	if (!(*addr))
+		return -FI_ENOMEM;
+
+	ret = sscanf(str, "%*[^:]://%" SCNx64 ":%" SCNx64,
+		     (uint64_t *) *addr, (uint64_t *) *addr + 1);
```

Comment:
This is fine, and is what several other providers do.  But does the address break into components that would be easier for a user to handle?  For example, see FI_SOCKADDR_IN, FI_SOCKADDR_IB, and FI_ADDR_IB_UD as an example for how the addressing can be broken into more decipherable pieces.

Reply:
Unfortunately, the 128 bit address is a combination items like in psm2 which are mostly opaque to the user.